### ISSUE TITLE
Compile Error with Clang 14 for Linux

### ISSUE THREAD

**Issue by Ravbug** [mikke89/RmlUi#545]
On Clang 14 for Linux, compiling RmlUi 5.1 fails with this error:
```cpp
In file included from /home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/RmlUi/Source/Core/Element.cpp:52:
/home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/RmlUi/Source/Core/ElementStyle.h:43:31: error: no type named 'uint8_t' in namespace 'std'; did you mean simply 'uint8_t'?
enum class PseudoClassState : std::uint8_t { Clear = 0, Set = 1, Override = 2 };
                              ^~~~~~~~~~~~
                              uint8_t
/usr/include/x86_64-linux-gnu/bits/stdint-uintn.h:24:19: note: 'uint8_t' declared here
typedef __uint8_t uint8_t;
                  ^
1 error generated.
```
Here is the full compile command line:
```
/usr/bin/clang++-14 -DRMLUI_ENABLE_LOTTIE_PLUGIN -DRMLUI_ENABLE_SVG_PLUGIN -DRMLUI_STATIC_LIB=1 -DRMLUI_VERSION=\"5.1\" -I/home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/RmlUi/Include -I/home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/rlottie/inc -I/home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/lunasvg/include -I/home/runner/work/RavEngine/RavEngine/build/deps/RmlUi-freetype/freetype/include -I/home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/freetype/include -I/home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/harfbuzz/src -O3 -DNDEBUG -w -MD -MT deps/RmlUi-freetype/CMakeFiles/RmlCore.dir/RmlUi/Source/Core/Element.cpp.o -MF deps/RmlUi-freetype/CMakeFiles/RmlCore.dir/RmlUi/Source/Core/Element.cpp.o.d -o deps/RmlUi-freetype/CMakeFiles/RmlCore.dir/RmlUi/Source/Core/Element.cpp.o -c /home/runner/work/RavEngine/RavEngine/deps/RmlUi-freetype/RmlUi/Source/Core/Element.cpp
```

Compilation succeeds on Apple Clang for Xcode 15, MSVC for UWP and Win32, and gcc for Linux.

**Comment by Ravbug** [comment]
Also reproduces on Clang 16

**Comment by mikke89** [comment]
Thanks for reporting, although I believe this one is already fixed on master: 313cbab6570a45fba1a08534da7c3b46b812fa48

**Comment by mikke89** [comment]
This reminded me that I'd like to add better CI coverage for newer GCC/clang, so I went ahead and added it now: 666690c5c5a9809fdc5ddf1a58adb54b40c5609c
